GLMakie/src/new-primitives.jl

```diff
@@ -746,8 +751,8 @@ end
 function draw_atomic(screen::Screen, scene::Scene, plot::LineSegments)
     attr = generic_robj_setup(screen, scene, plot)
 
-    add_input!(attr, :px_per_unit, screen.px_per_unit)
-    add_input!(attr, :viewport, scene.viewport)
+    haskey(attr, :px_per_unit) || add_input!(attr, :px_per_unit, screen.px_per_unit)
```

Comment:
related to the texture atlas comment, I think we may want to only add inputs in Makie.jl / Plot constructor.
ALthough screen couldn't be added there. 
But I still need to think about if we even should add the screen to the plot object.